### DISCUSSION TITLE
`Failed to execute C/C++ compiler` when building tree-sitter grammar

### DISCUSSION THREAD

**Discussion by hammadmajid** [helix-editor/helix#6925] [Q&A] [answered]
When I run the `hx --grammar build` command to build the tree-sitter grammars Helix throws the following error:

```
Building 142 grammars
142 grammars failed to build

    Failure 0/142: Failed to execute C/C++ compiler
    Failure 1/142: Failed to execute C/C++ compiler
    Failure 2/142: Failed to execute C/C++ compiler
    ...
    ...
    Failure 140/142: Failed to execute C/C++ compiler
    Failure 141/142: Failed to execute C/C++ compiler
```

**System specification:**

- Helix: 23.03 (3cf03723)
- GCC: 11.2.0
- G++: 11.2.0
- PowerShell: 7.3.4
- Windows: 22H2

**Comment by FlamongoBird** [comment]
Lol having the same issue... you didn't happen to find a solution did you?

**Comment by hammadmajid** [comment, marked as answer]
Sorry for the late reply but here is the link to stackoverflow question I posted. It has the solution I hope you are looking for
https://stackoverflow.com/questions/75820561/how-do-i-fix-failed-to-execute-c-c-compiler-error-when-building-helix-gramme/76923649#76923649

**Reply by FlamongoBird** [reply]
Thank you, I eventually got it figured out, and then did the same thing I always _don't_ want other people to do and didn't say how I fixed it lol. 

I tried installing gcc and g++ and that did not fix it, granted I really don't know what I'm doing on windows. What worked for me was just installing the C++ build tools with VScode, not sure why that worked but it did.

**Reply by hammadmajid** [reply]
I think installing gcc and g++ manually only adds them to **user path** but in **system path**. installing C++ build does add them to **system path**
Helix probably expect them in the **system path** (This is just a theory btw, I don't use Windows that much)
